drivers/hmis/app/graphql/sources/ce_step_referral_source.rb

```diff
@@ -0,0 +1,26 @@
+###
+# Copyright 2016 - 2025 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# frozen_string_literal: true
+
+module Sources
+  class CeStepReferralSource < GraphQL::Dataloader::Source
+    # Custom GraphQL Dataloader to resolve the relationship between steps and referrals.
```

Comment:
Nice!
Suggest clarifying comment to specify that this loader "fetches" `Hmis::WorkflowExecution::Step` records, and "resolves" the `Hmis::Ce::Referral` that each step belongs to. Not sure if I'm using the most accurate verbs there for Dataloader. But it would be helpful to clarify the input & output for future comprehension.

Reply:
Thank you for inspiring me to poke more around the [ruby graphql documentation for custom data sources](https://graphql-ruby.org/dataloader/sources.html). I ended up concluding that it might be more graceful to refactor this into a `CeReferralByInstanceIdSource`, so it accepts `instance_ids`, instead of `steps`. The documentation refers to the input as "keys", and having the keys be strings and not objects seems a little easier to follow logically, plus it may be useful in the future (for example, WFE audit events also store a reference to instance ID - though I don't know if we would ever resolve those outside the context of a referral).

I took a shot at the refactor and adding more clarifying comments. Let me know if you think it looks better!